packages/window-size-tool/src/engine.js

```diff
@@ -386,43 +383,42 @@ async function applyRule(
   let results = [];
   let hasPageChanges = false;
 
-  rules.forEach(async(rule, index) => { 
+
+  for(i=0; i<rules.length; i++){
       if (verbose) {
-        console.debug(`rule: ${chalk.yellow(index + 1)}`);
+        console.debug(`rule: ${chalk.yellow(i + 1)}`);
       }
-
-      const linkMatches = checkLinkWidth(pageNode, rule, verbose);
+      const linkMatches = checkLinkWidth(pageNode,rules[i], verbose);
       const hasLinks = linkMatches.length > 0;
-
-      if (checkPageWidth(pageNode, rule.width, verbose)) {
-        const pagePass = await checkRule(pageNode, filename, rule, verbose, domainsCheckEnabledForAllRules);
-        hasChanges = true;
-
-        if (pagePass) {        
+   
+      if(!hasChanges){
```

Comment:
Will this logic work in a scenario where there is a PAGE with WINDOW_OPTIONs which matches a rule, but also contains a LINK with WINDOW_OPTIONS to another page that matches a different rule?

Or, once the parent PAGE matches a rule, will the LINKed page stop getting evaluated against the subsequent rules? This might have been the reason the hasChanges check was removed?

e.g. something like… 
```
<PAGE id="parentPage" WINDOW_OPTIONS="width:900">
  <CLUSTER NUM_COLS="1">
  …
  </CLUSTER>
  <LINK PAGE_ID="linkedPage" WINDOW_OPTIONS="width:500"/>
</PAGE>
```

```
<PAGE id="linkedPage">
  <CLUSTER NUM_COLS="2">
  …
  </CLUSTER>
</PAGE>
```


In that example `parentPage` should be sized down to medium/700 and LINK to `linkedPage` should be sized up to medium/700 i.e. two different rules being applied within the same document. Just checking that this works?